ASM/c/dungeon_info.c

```diff
@@ -835,8 +854,23 @@ void draw_dungeon_info(z64_disp_buf_t* db) {
 
         for (int i = 0; i < 12; i++) {
             dungeon_entry_t* d = &(dungeons[i + (i > 9 ? 1 : 0)]); // skip Hideout
+            bool empty = CFG_DUNGEON_PRECOMPLETED[d->index];
             int top = start_top + ((icon_size + padding) * i) + 1;
-            text_print(db, d->name, left, top);
+            if (empty) {
+                gDPSetPrimColor(db->p++, 0, 0, 0xFF, 0xFF, 0xFF, 0x7F);
+                uint16_t sizeRectangle = text_print(db, d->name, left, top) - left;
+                gDPSetPrimColor(db->p++, 0, 0, 0xFF, 0xFF, 0xFF, 0xBF);
+                gDPSetCombineMode(db->p++, G_CC_PRIMITIVE, G_CC_PRIMITIVE);
+                gSPTextureRectangle(db->p++,
+                        left * 4, (top + 6) * 4,
+                        (left + sizeRectangle) * 4, (top + 6 + 2) * 4,
+                        0,
+                        0, 0,
+                        1024, 1024);
+                gDPSetCombineMode(db->p++, G_CC_MODULATEIA_PRIM, G_CC_MODULATEIA_PRIM);
+            } else {
+                text_print(db, d->name, left, top);
```

Comment:
Does this work properly? Without a `gDPSetPrimColor` won't this be whatever color happened to be set last?

Reply:
Indeed, colors were all wrong on this part 🥲